Dependency injection/Hilt.md

```diff
@@ -0,0 +1,178 @@
+  * `@Provides` - Place above the function in module that returns an instance of the dependency that you want to inject somewhere.
+  * `@Binds` - Also used in modules, but the function can be abstract or an interface function. Intead of creating and returning the instance, the function must return dependency type (usually an interface) and take the concrete implementation as a parameter.
+  * `@HiltViewModel` - Place above ViewModels to populate the `ViewModelComponent` enabling view models to be injected.
+
+
+### How it works
+
+In Hilt, everything happens through annotations. As I mentioned before, the first one you need to add is `@HiltAndroidApp` above your application class. This will generate all the components that you will need.
+
+Keep note of the scope of each generated component. For example, if some dependenci is used in ViewModels only, you need to install it in the `ViewModelComponent`.
+
+`@AndroidEntryPoint` is the next annotation that you will probably use the most. Placed above Activity or Fragment, it makes them viable for injection by adding them to Hilt generated builder modules. No further action required.
+
+`@HiltViewModel` does the same, just for view models. Once you add this annotation, you can feel free to `@Inject` dependencies to your view models. As long as dependencies are provided in some `@Module`.
+
+`@Module` and `@InstallIn` are the pair you will want to use with your modules. Every module consist of either `@Provides` or `@Binds` annotated functions that will provide dependencies as explained earlier.
+
+`@Inject` placed beside a dependency in the `constructor`, `field` or `method` will make sure that your dependency is injected where needed.
+
+Let's dive into a real project example to see how this all works together.
+
+### Example
+
+First we start with an application class:
+
+```kotlin
+@HiltAndroidApp
+class AliasApp : Application()
+```
+
+Then, define some modules that will provide dependencies:
+
+```kotlin
+@Module
+@InstallIn(SingletonComponent::class)
+class RemoteConfigModule {
+
+    @Provides
+    @Singleton
+    fun remoteConfig(): FirebaseRemoteConfig =
+        Firebase.remoteConfig
+}
+```
+
+```kotlin
+@Module
+@InstallIn(ViewModelComponent::class)
+interface RepositoriesModule {
+
+    @Binds
+    fun teamsRepo(teamsRepository: TeamRepositoryImpl): TeamRepository
+
+    @Binds
+    fun wordsRepo(wordsRepositoryImpl: WordRepositoryImpl): WordRepository
+
+    @Binds
+    fun myWordsRepo(wordsRepositoryImpl: MyWordRepositoryImpl): MyWordRepository
+
+    @Binds
+    fun databaseVersionRepo(databaseVersionRepositoryImpl: DatabaseVersionRepositoryImpl): DatabaseVersionRepository
+}
+```
+
+We are using `SingletonComponent` for the first module, because we need remote config in various places, so we are making sure it's available in the global scope. Also, by using `@Singleton` annotation, we are making sure that a single instance will be made and provided in all places.
```

Comment:
```suggestion
We are using the `SingletonComponent` for the first module, because we need remote config in various places, so we are making sure it's available in the global scope. Also, by using the `@Singleton` annotation, we are making sure that a single instance will be made and provided in all places.
```